nexus/types/src/external_api/views.rs

```diff
@@ -49,14 +49,64 @@ pub struct Silo {
         BTreeMap<shared::SiloRole, BTreeSet<shared::FleetRole>>,
 }
 
+/// A collection of resource counts used to describe capacity and utilization
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
-pub struct SiloQuotas {
-    pub silo_id: Uuid,
+pub struct VirtualResourceCounts {
+    /// Number of virtual CPUs
     pub cpus: i64,
+    /// Amount of memory in bytes
     pub memory: ByteCount,
+    /// Amount of disk storage in bytes
     pub storage: ByteCount,
 }
 
+/// A collection of resource counts used to set the virtual capacity of a silo
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct SiloQuotas {
+    pub silo_id: Uuid,
+    #[serde(flatten)]
+    pub limits: VirtualResourceCounts,
+}
+
+// For the eyes of end users
+/// View of the current silo's resource utilization and capacity
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct Utilization {
+    /// Accounts for resources allocated to running instances or storage allocated via disks or snapshots
+    /// Note that CPU and memory resources associated with a stopped instances are not counted here
+    /// whereas associated disks will still be counted
+    pub provisioned: VirtualResourceCounts,
```

Comment:
Are these supposed to have `#[serde(flatten)]` on them too? I think this is why `VirtualResourceCounts` is showing up in the spec. On the other hand it's kind of cool to not flatten it and share the structure, but it's possible the other language clients might not like it as much.

Reply:
Whoops, I see now why these are not. They're supposed to be under these keys. In any case this is still probably why `VirtualResourceCounts` is making it into the spec.